lib/galaxy/managers/jobs.py

```diff
@@ -121,6 +127,13 @@ def index_query(self, trans, payload: JobIndexQueryPayload) -> sqlalchemy.engine
         search = payload.search
         order_by = payload.order_by
 
+        if trans.user is None:
+            # If the user is anonymous we can only return jobs for the current session history
+            if trans.galaxy_session and trans.galaxy_session.current_history_id:
+                history_id = trans.galaxy_session.current_history_id
+            else:
+                return None
```

Comment:
I'm wondering if we should raise an `AuthenticationRequired` exception here instead, so the error is not hidden from the user.

Reply:
This is not an error, if a session doesn't have a history it also doesn't have any jobs.

Reply:
But None is returned also is `trans.galaxy_session` is None.

Reply:
You could as well query some public thing, like jobs belonging to a history, raising an exception here feels very unpredictable.

Reply:
Then I don't understand the changes in lines 131-133 above: if we want a session-less user to be able to view public history jobs, we should overwrite `history_id` only when it's `None` .

Reply:
That is a good point, we might want to instead filter on the job itself using the session id.

Reply:
If `history_id` is passed in the API payload, the job's history accessibility is checked on the jobs returned by this method at https://github.com/galaxyproject/galaxy/blob/7caee0b7511152937e51229c28cdb778824d122d/lib/galaxy/webapps/galaxy/services/jobs.py#L89

Reply:
Thanks for the catch! I'll follow up with the proposed change :+1: